src/static/containers/AdminCardAdd.js

```diff
@@ -67,82 +66,16 @@ class AdminCardAdd extends Component {
-        });
-    }
-
-    addSource = () => {
-        const { form } = this.props;
-        const sourceKeys = form.getFieldValue('sourceKeys');
-        const nextSourceKeys = sourceKeys.concat(sourceId);
-        sourceId += 1;
-        form.setFieldsValue({
-            sourceKeys: nextSourceKeys,
-        });
-    }
-
     render() {
-        const { isFetchingCard, form: { getFieldValue, getFieldDecorator } } = this.props;
-        getFieldDecorator('helpLinkKeys', { initialValue: [] });
-        const helpLinkKeys = getFieldValue('helpLinkKeys');
-        const helpLinksItems = helpLinkKeys.map(k => (
-            <Form.Item>
-                {getFieldDecorator(`help_links[${k}]`, {
-                    validateTrigger: ['onChange', 'onBlur'],
-                    rules: [{
-                        required: true,
-                        whitespace: true,
-                        message: 'Saisissez un lien.',
-                    }],
-                })(<Input placeholder="Lien" />)}
-                <Icon
-                    className="dynamic-delete-button"
-                    type="minus-circle-o"
-                    onClick={() => this.removeHelpLink(k)}
-                />
-            </Form.Item>
-        ));
-        getFieldDecorator('sourceKeys', { initialValue: [] });
-        const sourceKeys = getFieldValue('sourceKeys');
-        const sourcesItems = sourceKeys.map(k => (
-            <fieldset>
-                <legend>Source n°{k}</legend>
-                <Form.Item label="Nom">
-                    {getFieldDecorator(`card_stats.data_sources[${k}].name`, {
-                        rules: [{ required: true, message: 'Nom de la source.' }],
-                    })(<Input />)}
-                </Form.Item>
-                <Form.Item label="Lien">
-                    {getFieldDecorator(`card_stats.data_sources[${k}].link`, {
-                        rules: [{ required: true, message: 'Lien vers la source.' }],
-                    })(<Input />)}
-                </Form.Item>
-                <Form.Item>
-                    {getFieldDecorator(`card_stats.data_sources[${k}].status`, {
-                        initialValue: 'UNVERIFIED',
-                    })(
-                        <Radio.Group>
-                            <Radio value="VERIFIED">Vérifiée</Radio>
-                            <Radio value="UNVERIFIED">Non vérifiée</Radio>
-                        </Radio.Group>
-                    )}
-                </Form.Item>
-            </fieldset>
-        ));
+        const { isFetchingCard, form } = this.props;
+        let cardForm;
+        // Update when we link the fetch!
+        const initialData = {};
+        if (this.state.editing) {
+            cardForm = generateForm(form, this.createCard, cardFields, initialData, 'Edit card!');
+        } else {
+            cardForm = generateForm(form, this.createCard, cardFields, initialData, 'Create card!');
+        }
```

Comment:
const cardForm = generateForm(form, this.createCard, cardFields, initialData, (this.state.editing) ? 'Éditer' : 'Créer');

Like this? More concise and clearer since there is no logic change, just text

Reply:
Maybe for now...
But, in the end, we maybe must use another function between creating a card and editing it and the initialData won't be the same (here, the editing is not working yet, but it wasn't before the PR!).
There still are some works to do, to make Card Form OK.

Reply:
ok